### PR TITLE
feat: refactor service mock with go

### PR DESCRIPTION
**Pull request by ev1lQuark**
## What is the purpose of the change

refactor service mock with go

## Brief changelog

- update gitignore for vscode config dir
- add mock service CRUD methods
- add mock service unit test

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist.
* [ ] Run `mvn clean compile --batch-mode -DskipTests=false -Dcheckstyle.skip=false -Drat.skip=false -Dmaven.javadoc.skip=true to make sure basic checks pass.

### CODE REVIEW

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
Please resolve conflicts

File: pkg/admin/services/registry_service_sync.go [hunk 1]
```diff
@@ -38,9 +38,9 @@ var (
 
 func init() {
 	queryParams := url.Values{
-		constant.InterfaceKey:  {constant.AnyValue},
-		constant.GroupKey:      {constant.AnyValue},
-		constant.VersionKey:    {constant.AnyValue},
+		constant.InterfaceKey:  {"org.apache.dubbo.springboot.demo.DemoService"},
+		constant.GroupKey:      {""},
+		constant.VersionKey:    {""},
```
Comment (AlbumenJ) [inline pkg/admin/services/registry_service_sync.go]: Why change this?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]
@zhaoyunxing92 @chickenlj PTAL

File: pkg/admin/model/mock_rule.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		Enable:      m.Enable,
+	}
+}
+
+type MockRuleEntity struct {
+	gorm.Model
+	ServiceName string `gorm:"type:varchar(255)"`
+	MethodName  string `gorm:"type:varchar(255)"`
+	Rule        string `gorm:"type:text"`
+	Enable      bool
```
Comment (chickenlj) [inline pkg/admin/model/mock_rule.go:49]: Is this definition the same as the original one listed here https://cn.dubbo.apache.org/zh-cn/overview/reference/admin/mock/#mock-rule?

We need to make sure previous users can smoothly migrate to the new version using the same backend MySQL server.

Comment (chickenlj) [reply pkg/admin/model/mock_rule.go:49]: Make sure to use the same schema

Comment (chickenlj) [reply pkg/admin/model/mock_rule.go:49]: Also, we should make sure new users know how to init the schema.

Comment (ev1lQuark) [reply pkg/admin/model/mock_rule.go:49]: I think GORM AutoMigrate function can handle above possible compatibility issues. I used this function in `/pkg/admin/config/config.go:200`
[GORM-AutoMigrate](https://gorm.io/zh_CN/docs/migration.html#AutoMigrate)

File: pkg/admin/services/registry_service_sync.go (hunk 1 above)
Comment (ev1lQuark) [reply pkg/admin/services/registry_service_sync.go]: For integration testing and I forget to reset them, I will update this PR.

Comment (ev1lQuark) [reply pkg/admin/services/registry_service_sync.go]: done

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**